### PR TITLE
Set the kubernetes version on rkeConfig spec during cluster create/update

### PR DESCRIPTION
**Pull request by prachidamle**
Enusre that the kubernetes version on rkeConfig spec is set during cluster create/update. If the client passed it as empty, rancher server sets it to system default k8s version.

This step ensures that during upgrade of rancher server if the system default k8s version is updated, existing clusters' k8s version does not get auto-upgraded.

https://github.com/rancher/rancher/issues/14407

@alena1108 need a review.

### CODE REVIEW

File: pkg/api/store/cluster/cluster_store.go [hunk 1]
```diff
@@ -88,3 +95,16 @@ func canUseClusterName(apiContext *types.APIContext, requestedName string) error
 
 	return nil
 }
+
+func setKubernetesVersion(data map[string]interface{}) {
+	rkeConfig, ok := values.GetValue(data, "rancherKubernetesEngineConfig")
```
Comment (alena1108) [inline pkg/api/store/cluster/cluster_store.go]: there is a way to get a nested value using values, I believe method is called GetValueN

File: pkg/controllers/management/clusterprovisioner/provisioner.go [hunk 2]
```diff
@@ -367,6 +367,7 @@ func getSystemImages(spec v3.ClusterSpec) (*v3.RKESystemImages, error) {
 		return nil, err
 	}
 
+	//version is set at API level during cluster create/update, but checking for empty version for backward compatability
```
Comment (alena1108) [inline pkg/controllers/management/clusterprovisioner/provisioner.go]: been thinking that for backwards compatibility, we should do something like:

```
version := spec.RancherKubernetesEngineConfig.Version
if version == "" {
//try getting it from the applied spec (if appliedSpec is not nil). Only if not there, fetch from settings
}
```
That would prevent upgrade for the cluster that was created w/o k8s version prior to this fix

File: pkg/api/store/cluster/cluster_store.go (hunk 1 above)
Comment (prachidamle) [reply pkg/api/store/cluster/cluster_store.go]: done

**Comment by alena1108** [comment]
LGTM

**Pull request merged**